### ISSUE TITLE
Not supported in Playnite 9

### ISSUE THREAD

**Issue by ritzbitz00** [FiercePunchStudios/HumbleKeysLibrary#2]
Playnite 9 changed a lot of things, this extension can't load as a result. Will it be updated for Playnite 9 compatibility?

**Comment by ritzbitz00** [comment]
Following up on this. Looking forward to the update once its ready, I have a lot of steam keys!

**Comment by manum45** [comment]
Thanks for this extension. Do you have a plan when it will be ready?

**Comment by JustinHardage** [comment]
Hi all, thanks for your patience! I've been swamped with day job stuff for about six months continuously, but I haven't forgotten about you.

I have a build that works with the new version of the SDK (6.2.2) in Playnite 9, except the settings page. Once I figure out how those settings changes work, I'll upload that build. If it goes smoothly, it'll be as soon as next weekend.

In the meantime, Crow added the ability for the main Humble library importer to detect keys, if you uncheck the checkbox for `Ignore third party store games`. Unfortunately, that importer doesn't tag the keys any differently than full games, and I haven't found any other way to distinguish those key entries. If you can figure out how to tell them apart, I could even retire this plug-in in favor of Crow's builtin extensions.

![image](https://user-images.githubusercontent.com/2737482/171020779-b8f6af79-fc89-441c-814b-301fe8d96ac4.png)

**Comment by ritzbitz00** [comment]
Thanks! My recollection is that the functionality of that setting didn't work as expected, which is why I had taken interest in your project in the first place.

https://www.reddit.com/r/playnite/comments/mv4d1q/humble_library_extension_bug_ignore_third_party/
https://www.reddit.com/r/playnite/comments/g8tynj/humble_ignore_3rd_party_games_not_working_as/

**Comment by JustinHardage** [comment]
I got the rest of this working on the weekend, but in full testing it turns out the Humble API is absolutely wilding. They return JSON with some values that can vary between string and object types. Which means Newtonsoft.Json throws errors when trying to deserialize it if Humble sends an unexpected type. I don't recall this being the case when I made the plugin last year.

The built in Humble extension deals with this by simply not acknowledging the existence of the key redemption value. But I have to check it to tag with the Redeemed / Unredeemed tags.

I expect to find a workaround in the next week. Fingers crossed that's the only big surprise 🤞
